### PR TITLE
Eperez actions

### PR DESCRIPTION
**Pull request by enriquepablo**
IdP actions.
Does not modify previous functionality, so should be safe to merge.
The git conflict is in the exceptions module.
The change in the MongoTemporaryInstance is to use it in the IdP tests, that had a fixed port.

### CODE REVIEW

File: src/eduid_userdb/actions/db.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+            return session
+        return userid + session
+
+    def clean_cache(self, userid, session=None):
+        '''
+        Delete cache for userid and IdP session.
+        Called at the start of a session to clean up
+        stale caches.
+
+        :param user: The user with possible pending actions
```
Comment (fredrikt) [inline src/eduid_userdb/actions/db.py]: user or userid? User() or str()?

**Comment by enriquepablo** [comment]
Fixed the docstring in ActionDB.clean_cache

File: src/eduid_userdb/actions/action.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        """"
+        Part of __init__().
+
+        Check actions that can't be loaded for some known reason.
+        """
+        if self._data_in.get('user_oid') is None:
+            raise ActionMissingData('Action {!s} has no user_oid'.format(
+                self._data_in.get('_id')))
+        if self._data_in.get('action') is None:
+            raise ActionMissingData('Action {!s} has no action'.format(
```
Comment (Gijutsu) [inline src/eduid_userdb/actions/action.py]: Maybe it is possible to formulate the message that goes with ActionMissingData a bit differently? Or maybe I am misunderstanding something about 'Action {!s} has no action'?

Comment (johanlundberg) [inline src/eduid_userdb/actions/action.py]: As discussed afk, maybe change the message to "Action {!s} has no key user_oid" and "Action {!s} has no key action".

Comment (enriquepablo) [inline src/eduid_userdb/actions/action.py]: Yes, Action has no action sounds weird, I have committed the change suggested by Johan, directly in master.

**Pull request merged**